test/lib/send_with_us/api_request_test.rb

```diff
@@ -68,14 +53,29 @@ def test_send_with_connection_refused
     assert_raises( SendWithUs::ApiConnectionRefused ) { @request.post(:send, @payload) }
   end
 
-  def test_send_with_version
+  def test_send_with_with_attachment
+    build_objects
+    email_id = 'test_fixture_1'
+    result = @api.send_with(
+      email_id,
+      {name: 'Ruby Unit Test', address: 'matt@example.com'},
+      {data: 'I AM DATA'},
+      {name: 'sendwithus', address: 'matt@example.com'},
+      [],
+      [],
+      ['README.md']
+    )
+    assert_instance_of( Net::HTTPOK, result )
```

Comment:
Not really. These requests actually hit sendwithus, while many of the other tests in this spec stub out Net::HTTP. Instead we should be testing that a request is being made with the correct payload and stub the response. We could have an integration test to determine the actual validity of the request when hitting the server that wouldnt always have to run with the test suite.

Moreso, we should only be testing methods related to `SendWithUs::ApiRequest` in this spec, and the spec should not have to be coupled to SendWithUs::Api in any way. There are three public methods that need to be tested and tests already exist for those. Instead we should test these details in the api spec and verify that SendWithUs::ApiRequest is receiving the correct payload.

I wouldnt ask that you fix all of the specs, but we should probably stop adding to the ones we should revisit because I can add or remove any of that data except for email_id and the "to" hash and still have that test pass

``ruby
[3] pry(#<TestApiRequest>)> @api.send_with(email_id, {name: "lol", address: "lol@lol.ca"})
=> #<Net::HTTPOK 200 OK readbody=true>
```